### PR TITLE
plumb through identifyBehavior

### PR DESCRIPTION
**Pull request by songgao**
r? @mmaxim Thanks!

This needs to ship with https://github.com/keybase/kbfs/pull/535; otherwise it crashes the KBFS process.

Dear GitHub please close #4812 when this is merged.

### CODE REVIEW

**Comment by songgao** [comment]
(I did some hacks to try the GUI mode from CLI, and it seems to be working. i.e. no pop-ups with GUI mode)

**Comment by mmaxim** [comment]
A couple high level comments:

1.) Can we change `TLFUserBreak` to be a more accurate name? Like `TLFIdentifyFailure`? The term break could mean anything.
2.) Is it possible to not create `ConversationLocalWithBreaks`? It just seems like an awkward name. Can we just have the "breaks" (which we should rename) sit along side the conversationsl ike we do in the result from `GetThreadLocal`?

File: go/chat/inboxsource.go [hunk 1]
```diff
@@ -16,13 +16,17 @@ import (
 
 type Inbox struct {
 	Version    chat1.InboxVers
-	Convs      []chat1.ConversationLocal
+	Convs      []chat1.ConversationLocalWithBreaks
 	Pagination *chat1.Pagination
```
Comment (mmaxim) [inline go/chat/inboxsource.go]: In other words, could we just have an additional field here for identify failures?

File: go/chat/inboxsource.go [hunk 2]
```diff
@@ -219,77 +242,85 @@ func (s *RemoteInboxSource) localizeConversationsPipeline(ctx context.Context, c
...
 func (s *RemoteInboxSource) localizeConversation(
-	ctx context.Context, conversationRemote chat1.Conversation) (
-	conversationLocal chat1.ConversationLocal, err error) {
+	ctx context.Context, conversationRemote chat1.Conversation, identifyBehavior keybase1.TLFIdentifyBehavior) (
+	conversationLocalWithBreaks chat1.ConversationLocalWithBreaks, err error) {
 
 	ctx, uimap := utils.GetUserInfoMapper(ctx, s.G())
 
-	conversationLocal.Info = chat1.ConversationInfoLocal{
+	conversationLocalWithBreaks.ConversationLocal.Info = chat1.ConversationInfoLocal{
```
Comment (mmaxim) [inline go/chat/inboxsource.go]: This is also super verbose, if we could reduce the length of these names it would be great.

**Comment by songgao** [comment]
Thanks @mmaxim !

1) I'm OK with changing `TLFUserBreak` to `TLFIdentifyFailure`. It would also change the KBFS side though. I'll make the PR;

2) The `ConversationLocalWithBreaks` is also the type returned by service RPCs. For example, `GetInboxAndUnboxLocalRes` has an array of conversations, and it'd be useful to group them together. The breaks cannot just go inside the `ConversationLocal` because it's the type that gets stored in local storage (in the future anyways). But I guess we could 1) in `chat_local.go`, i.e. service, shorten the name, e.g. how about `CLB` as a record consisting of `ConversationLocal` and `[]TLFIdentifyFailure`?; 2) in `inboxsource.go`, just expand it into two fields like what you suggested. Does it sound good?

**Comment by mmaxim** [comment]
I thought you could just do for the inbox res what you currently do for the thread view result:

```
   record GetThreadLocalRes {
      ThreadView thread;		     
      array<RateLimit> rateLimits;		     
      array<keybase1.TLFUserBreak> breaks;
    }		    
```

Just stick it alongside in the result type.

**Comment by mmaxim** [comment]
Also I am wondering if we need to make the identify behavior a parameter to all these RPCs. Could we just figure out which behavior to use based on the client type on the connection?

**Comment by songgao** [comment]
> Just stick it alongside in the result type.

but for `GetInboxAndUnboxLocalRes`, it would become:
```
  record GetInboxAndUnboxLocalRes {
    array<ConversationLocalWithBreaks> conversations;
    union { null, Pagination } pagination;
    array<RateLimit> rateLimits;
    array<array<keybase1.TLFUserBreak>> breaks;
  }
```
which isn't very nice...

> Also I am wondering if we need to make the identify behavior a parameter to all these RPCs. Could we just figure out which behavior to use based on the client type on the connection?

Sounds interesting! That eliminates the possibility to use GUI mode from `kbc api` though. But I guess that's probably fine?

**Comment by mmaxim** [comment]
Why would `ConversationLocalWithBreaks` need to continue existing? Wouldn't it be:

```
 record GetInboxAndUnboxLocalRes {
    array<ConversationLocal> conversations;
    union { null, Pagination } pagination;
    array<RateLimit> rateLimits;
    array<array<keybase1.TLFUserBreak>> breaks;
  }
```

**Comment by mmaxim** [comment]
So maybe the behavior parameter on the RPCs is like an override kind of thing, might make sense to ask Nojima what would be most convenient on their side.

**Comment by songgao** [comment]
> Why would ConversationLocalWithBreaks need to continue existing? Wouldn't it be:

Yes sorry I copied and pasted the chunk but forgot to change the type. But still, it has an array of arrays of breaks, and the caller would need to use index to associate the breaks with conversations.

**Comment by mmaxim** [comment]
Could we make it a pointer inside `ConversationLocal` that is just `nil` when we write to local storage? We aren't even using that inbox storage right now, I'd hate to see it be the source of making things more complicated outside of it.

**Comment by songgao** [comment]
@mmaxim I guess that would work. But I'm worried it would cause confusion. If you get a `ConversationLocal` with `nil` break, does it mean it identified with no breaks, or does it mean it doesn't have information on identify?

**Comment by mmaxim** [comment]
Or how about we store a different type to storage then, call it `ConversationStorage` and not have the identify break list in there.

**Comment by songgao** [comment]
Hmm that's interesting. So we can only have a `ConversationLocal` that has run the identify, and if the breaks is `nil`, there's no break. Sounds good to me!

**Comment by mmaxim** [comment]
Yeah `ConversationLocal` just has the list of identify breaks in it all the time, and this new type `ConversationStorage` does not.

**Comment by songgao** [comment]
Sounds good; thanks Mike!

**Comment by songgao** [comment]
@mmaxim I'm not sure how to detect connection type and whether it's necessary for now, but other parts are ready for review. Thanks!

sister PR in KBFS: https://github.com/keybase/kbfs/pull/543

**Comment by keybase-ci-visdiff** [comment]
The commits 02c542268044...a3432b7cabd0 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480462345672-02c542268044-a3432b7cabd0-linux/Paper_Key-Normal-02c542268044.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480462345672-02c542268044-a3432b7cabd0-linux/Paper_Key-Normal-a3432b7cabd0.png)  
   Paper_Key-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 41edef05f823...f08654d58eed introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Pinentry-Paper_Key_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480537026513-41edef05f823-f08654d58eed-linux/Pinentry-Paper_Key_Error-41edef05f823.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480537026513-41edef05f823-f08654d58eed-linux/Pinentry-Paper_Key_Error-f08654d58eed.png)  
   Pinentry-Paper_Key_Error comparison
</summary>

**Pull request merged**